### DISCUSSION TITLE
Automatically add by default to "dependencies" or "devDependencies" depending on the dependency

### DISCUSSION THREAD

**Discussion by adriancuadrado** [npm/rfcs#429] [RRFC]
It is very easy to forget using the `--save-dev` flag to add a dependency correctly. I was thinking it would be a good idea if the package to add had in its package.json a field to tell whether this package is meant to be added as a normal dependency or as a devDependency. This way you won't have to explicitly add the `--save-dev` flag to add the dependency as a devDependency anymore. This new field is optional and its default value will make npm install add the dependency to the `dependencies` field.

**Comment by ljharb** [comment]
There's no way for the package author to know that. `eslint`, for example, is usually a dev dep, but it's a peer dep for plugins, and could be a runtime dep of a wrapper binary.

You always have to know, and explicitly indicate, when something isn't a runtime dependency.

**Reply by adriancuadrado** [reply]
What if the dependency is a test framework? Also, you always might have to know, but if in 99% of the cases a dependency is added as a devDependency, then it might be worth it. Besides, adding this feature doesn't mean making the user unable to explicitly download the dependency as a normal or dev dep. Just like now you can add to devDependencies using --save-dev, maybe we could add another flag to explicitly add as a normal dep the dependency you want to add to override the default.

**Reply by ljharb** [reply]
Something that is wrong *ever* is not likely to be worth it.

`--save-prod` and `--save-peer` and `--save-optional` already exist, so defaulting to dev would still allow those flags to overwrite - the issue remains that the package author simply can never know for sure if a dep is a dev, prod, peer, or optional dep.
